src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T29MWTier.tsx

```diff
@@ -0,0 +1,200 @@
+      Events.heal.by(SELECTED_PLAYER).spell(TWO_PIECE_SPELLS),
+      this.handle2PcHeal,
+    );
+    if (this.has4Piece) {
+      this.addEventListener(
+        Events.heal.by(SELECTED_PLAYER).spell(FOUR_PIECE_SPELLS),
+        this.handle4PcHeal,
+      );
+      this.addEventListener(
+        Events.heal.by(SELECTED_PLAYER).spell(SPELLS.VIVIFY),
+        this.handle4PcVivify,
+      );
+      this.addEventListener(
+        Events.heal.by(SELECTED_PLAYER).spell(SPELLS.RENEWING_MIST_HEAL),
+        this.handleRemTick,
+      );
+    }
+  }
+
+  get total4PieceHealing() {
+    return this.fourPieceHealingFromBuff + this.extraRemHealing + this.extraVivHealing;
+  }
+
+  handleEfBolt(event: ApplyBuffEvent) {
+    const playerId = event.targetID;
+    if (
+      !this.hotTracker.hots[playerId] ||
+      !this.hotTracker.hots[playerId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    this.hotTracker.hots[playerId][
+      SPELLS.RENEWING_MIST_HEAL.id
+    ].maxDuration! += FOUR_PIECE_EXTENSION;
+    this.hotTracker.addExtension(
+      EXTENSION_ATTRIB,
+      FOUR_PIECE_EXTENSION,
+      playerId,
+      SPELLS.RENEWING_MIST_HEAL.id,
+      event.timestamp,
+    );
+
+    this.numExtensions += 1;
+    this.totalExtensionDuration += this.hotTracker.hots[playerId][
+      SPELLS.RENEWING_MIST_HEAL.id
+    ].extensions.at(-1)!.amount;
+  }
+
+  handle2PcHeal(event: HealEvent) {
+    if (
+      !this.hotTracker.hots[event.targetID] ||
+      !this.hotTracker.hots[event.targetID][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    this.twoPieceHealing += calculateEffectiveHealing(event, TWO_PIECE_BONUS);
+  }
+
+  handle4PcHeal(event: HealEvent) {
+    this.fourPieceHealingFromBuff += calculateEffectiveHealing(event, FOUR_PIECE_BONUS);
+  }
+
+  handle4PcVivify(event: HealEvent) {
+    const targetId = event.targetID;
+    if (
+      !this.hotTracker.hots[targetId] ||
+      !this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id];
+    const extensionForVivify = this.hotTracker.getRemExtensionForTimestamp(hot, event.timestamp);
+    if (extensionForVivify?.attribution.name.startsWith(ATTRIBUTION_PREFIX)) {
+      this.extraVivHealing += (event.amount || 0) + (event.absorbed || 0);
+    }
+  }
+
+  handleRemTick(event: HealEvent) {
+    const targetId = event.targetID;
+    if (
+      !this.hotTracker.hots[targetId] ||
+      !this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id];
+    const extension = this.hotTracker.getRemExtensionForTimestamp(hot, event.timestamp);
+    if (extension?.attribution.name.startsWith(ATTRIBUTION_PREFIX)) {
+      this.extraRemHealing += (event.amount || 0) + (event.absorbed || 0);
+    }
+
+    if (HasRelatedEvent(event, 'FromMistyPeaks')) {
```

Comment:
either add a debug bool or remove

Reply:
Oops yea, need to remove that. did not mean to include that in the pr